lib/derivations.nix

```
@@ -206,4 +206,40 @@ in
   optionalDrvAttr =
     cond:
     value: if cond then value else null;
+
+  /**
+    Wrap a derivation such that instantiating it produces a warning.
+
+    # Inputs
+
+    `msg`
+    : The warning message to emit (via lib.warn).
+
+    `pkg`
+    : The derivation to wrap.
+
+    # Examples
+    :::{.example}
+    ## `lib.derivations.wrapWithWarn` usage example
+
+    ```nix
+    {
+      myPackage = wrapWithWarn "myPackage has been renamed to my-package" my-package;
+    }
+    ```
+
+    :::
+  */
+  wrapWithWarn =
+    msg: pkg:
```

Since package is not as well defined as a derivation is. And you also call it 'The derivation to wrap.' in your docs.
```suggestion
    msg: drv:
```

Or did you really mean package? This includes derivations that are built by `mkDerivation` rather than only `builtins.derivation`

---

I think it's good to talk about these things as packages at the language level, because of the [package attrset definition](https://nix.dev/manual/nix/2.24/glossary#package-attribute-set), but until packages and derivations are untangled in more places I don't expect anyone to get it right anyway, if there even is such a thing as right (before more untangling, e.g. https://github.com/NixOS/nix/issues/6507, Nixpkgs code for rfc 92 dynamic derivations, etc).
(Also let's not make a fuss about it until that stuff is more relevant in Nixpkgs. I'm already oversharing.)

---

Changed to `drv` – may as well be consistent.